pkg/volume/glusterfs/glusterfs.go

```diff
@@ -1045,3 +1056,62 @@ func parseClassParameters(params map[string]string, kubeClient clientset.Interfa
 	}
 	return &cfg, nil
 }
+
+type glusterfsDiskExpander struct {
+	*glusterfs
+	provisionerConfig
+	spec *v1.PersistentVolume
+}
+
+var _ volume.Expander = &glusterfsDiskExpander{}
+
+func (e *glusterfsDiskExpander) ExpandVolumeDevice(newSize resource.Quantity, oldSize resource.Quantity) error {
+
+	pvSpec := e.spec.Spec
+	glog.V(2).Infof("Request to expand volume: %s ", pvSpec.Glusterfs.Path)
+	volumeName := pvSpec.Glusterfs.Path
+
+	// Fetch the volume for expansion.
+	volumeID := dstrings.TrimPrefix(volumeName, volPrefix)
+
+	//Get details of SC.
+	class, err := volutil.GetClassForVolume(e.plugin.host.GetKubeClient(), e.spec)
+	if err != nil {
+		return err
+	}
+	cfg, err := parseClassParameters(class.Parameters, e.plugin.host.GetKubeClient())
+	if err != nil {
+		return err
+	}
+	e.provisionerConfig = *cfg
+
+	glog.V(4).Infof("Expanding volume %q with configuration %+v", volumeID, e.provisionerConfig)
+
+	//Create REST server connection
+	cli := gcli.NewClient(e.url, e.user, e.secretValue)
+	if cli == nil {
+		glog.Errorf("failed to create glusterfs rest client")
+		return fmt.Errorf("failed to create glusterfs rest client, REST server authentication failed")
+	}
+
+	// Find out delta size
+	expansionSize := int(newSize.Value() - oldSize.Value())
```

Comment:
Is the `oldSize` guaranteed to be accurate? If not, your delta could be wrong.

Reply:
`oldSize` is picked from `pvc.Status.Capacity`  and since there can be only one resize operation active at a time for a volume, it is guaranteed to be correct. If we were to pick it from `pvc.Spec.Resources.Requests["Storage"]` or something - it would be inaccurate, but reading from `Status` should give us correct result.

Reply:
Yep. thats what my understanding too